src/lake/Lake/Config/LeanConfig.lean

```diff
@@ -157,10 +157,17 @@ structure LeanConfig where
   They come *before* `moreLinkArgs`.
   -/
   weakLinkArgs : Array String := #[]
-
   /--
-    Compiler backend that modules should be built using (e.g., `C`, `LLVM`).
-    Defaults to `C`.
+  Compiler backend that modules should be built using (e.g., `C`, `LLVM`).
+  Defaults to `C`.
   -/
   backend : Backend := .default
+  /--
+  Asserts Lake should assume the Lean code is platform-independent.
+  This means Lake will not include platform-dependent elements (such as
+  shared libraries) in a module's trace and will not re-elaborate Lean code
+  for different platforms.
+  Defaults to `false`.
+  -/
+  platformIndependent : Bool := false
```

Comment:
From the name and description I would expect .oleans even without shared libraries to get platform-specific traces by default but I don't think that's the case, is it? Should it be?

Reply:
No, it is not the case and I do not think it should be (at least as the default). It was not my intention to convey that with the docstring. What gave you that impression? Would "This means Lake will exclude platform-independent elements (such as shared libraries) from the module's trace (if they exist)" be better?

Reply:
I don't know how to interpret "the Lean code is platform-independent" then. If the *compile-time* code is platform-dependent, it seems like the .olean hash should depend on the platform under any configuration. Asked differently, why exactly should setting `precompileModules` change the default assumption to "platform-dependent" when no external libraries are involved?

Reply:
I agree with your assessment that "If the output of compile-time code is platform-dependent, it seems like the .olean hash should depend on the platform under". The difficulty is that it is hard for Lake to know whether this is the case. Currently, Lake assumes not unless one of the inputs is naturally platform-dependent (e.g. a shared library). 

What I think you're saying here is that, with this option, we can avoid assumptions and just operate based on this flag? That is, if it is `false`, we add the platform to the module trace. Otherwise, we exclude platform-dependent hashes from the trace. Is that right?

Reply:
I have updated the `platformIndepedent` configuration to a ternary value. `none` will retain the pre-PR behavior, `false` will add the platform to the trace, and `true` will exclude native objects from the trace. I have also ensured that the Lake-provided native objects always include the platform in their trace. Hopefully, this will address this concern.

Reply:
Sounds exhaustive, great!